### PR TITLE
Add language fallback option

### PR DESCRIPTION
**Pull request by patrickniepel**
- Add language fallback option

### CODE REVIEW

File: Sources/RswiftCore/Generators/StringsStructGenerator.swift [hunk 1]
```diff
@@ -248,13 +248,20 @@ struct StringsStructGenerator: ExternalOnlyStructGenerator {
...
         )
       ],
       doesThrow: false,
       returnType: Type._String,
       body: """
         guard let preferredLanguages = preferredLanguages else {
-          return \(values.swiftCode(bundle: "hostingBundle"))
+          let localizedString = \(values.swiftCode(bundle: "hostingBundle"))
+
+          \(values.swiftFallbackCodeNoParams(bundle: "hostingBundle"))
```
Comment (EmDee) [inline Sources/RswiftCore/Generators/StringsStructGenerator.swift]: `return` value missing here or what does `localizedString` do?

File: Sources/RswiftCore/Generators/StringsStructGenerator.swift [hunk 2]
```diff
@@ -298,7 +312,9 @@ struct StringsStructGenerator: ExternalOnlyStructGenerator {
       returnType: Type._String,
       body: """
         guard let preferredLanguages = preferredLanguages else {
-          let format = \(values.swiftCode(bundle: "hostingBundle"))
+          let localizedString = \(values.swiftCode(bundle: "hostingBundle"))
```
Comment (EmDee) [inline Sources/RswiftCore/Generators/StringsStructGenerator.swift:319]: I'm confused by this `localizedString`. Where is it used?

File: Sources/RswiftCore/Generators/StringsStructGenerator.swift [hunk 3]
```diff
@@ -387,4 +403,22 @@ private struct StringValues {
 
     return results
   }
+
+  func swiftFallbackCodeNoParams(bundle: String) -> String {
+    let escapedKey = key.escapedStringLiteral
+
+    return """
+    guard localizedString == \"\(escapedKey)\", let fallbackBundlePath = \(bundle).path(forResource: fallbackLanguage, ofType: "lproj"), let fallbackLocalizationBundle = Bundle(path: fallbackBundlePath) else { return localizedString }
```
Comment (patrickniepel) [inline Sources/RswiftCore/Generators/StringsStructGenerator.swift]: `localizedString` is used here for example. I've already tested this btw. It compiles and it works 😄

File: Sources/RswiftCore/Generators/StringsStructGenerator.swift (hunk 2 above)
Comment (patrickniepel) [reply Sources/RswiftCore/Generators/StringsStructGenerator.swift:319]: See https://github.com/pace/R.swift/pull/1#discussion_r798313328

File: Sources/RswiftCore/Generators/StringsStructGenerator.swift (hunk 1 above)
Comment (patrickniepel) [reply Sources/RswiftCore/Generators/StringsStructGenerator.swift]: See https://github.com/pace/R.swift/pull/1#discussion_r798313328

File: Sources/RswiftCore/Generators/StringsStructGenerator.swift (hunk 2 above)
Comment (patrickniepel) [reply Sources/RswiftCore/Generators/StringsStructGenerator.swift:319]: It is indeed confusing because it's Swift code that generates Swift code

Comment (EmDee) [reply Sources/RswiftCore/Generators/StringsStructGenerator.swift:319]: Strange. Can you show me how the generated code looks like (just out of interest)?

Comment (patrickniepel) [reply Sources/RswiftCore/Generators/StringsStructGenerator.swift:319]: Screenshot 2022-02-03 at 09 31 54

**Comment by patrickniepel** [comment]
@EmDee Might need to do another change

**Comment by patrickniepel** [comment]
@EmDee Done here. I've cleaned the code up, should be less confusing now

File: Sources/RswiftCore/Generators/StringsStructGenerator.swift (hunk 2 above)
Comment (EmDee) [reply Sources/RswiftCore/Generators/StringsStructGenerator.swift:319]: Yeah, that makes more sense now.

**Pull request merged**